ocaml/xapi/xapi_clustering.ml

```diff
@@ -181,29 +181,30 @@ let get_corosync_version () =
     |> handle_error
 
 let maybe_switch_cluster_stack_version ~__context ~cluster_stack_version =
-  match Context.get_test_clusterd_rpc __context with
-  | Some _ ->
-      debug "in unit test, not switching cluster stack verion"
-  | None -> (
-      if cluster_stack_version = 3L then
-        match get_corosync_version () with
-        | Cluster_stack.Corosync2 -> (
-          try
-            let out, _err =
-              Forkhelpers.execute_command_get_output
-                "/usr/libexec/set-system-corosync-version" ["3"]
-            in
-            debug "%s: switched to corosync3 %s" __FUNCTION__ out
-          with Forkhelpers.Spawn_internal_error (out, err, _status) ->
-            Unix_error
-              (Printf.sprintf
-                 "failed to switch corosync version out: %s, err :%s" out err
-              )
-            |> handle_error
-        )
-        | _ ->
-            debug "already running corosync3, no need to switch"
-    )
+  if Xapi_cluster_helpers.corosync3_enabled ~__context then
```

Comment:
Ah. I see. This feature flag is only for the function `switching an existing cluseter from corosync2 to corosync3`. A user can still create a new cluster with corosync3 without caring about this new feature flag.

Reply:
Yep! This feature is to restrict the upgrade process, not restricting the feature itself. A user would be able to create a corosync2 cluster as usual in XS 8.